### PR TITLE
[DESIGN2-201-Checkbox] Checkbox Component - DSCO - Update color variables

### PR DESCRIPTION
**Pull request by kelbyhawn**
Swaps out color variables from `shared/color.scss` with Primitive colors from `primitiveColors.css` on the Checkbox component. 

👀 **DOTD note:** there might be Eyes diffs with this.

## Links
Jira ticket: [DESIGN2-201](https://codedotorg.atlassian.net/browse/DESIGN2-201)

## Testing story
Tested locally in Storybook

----

## Before - using `shared/colors.scss`
Before

## After - using `primitiveColors.css`
After

### CODE REVIEW

File: apps/src/componentLibrary/checkbox/checkbox.module.scss [hunk 1]
```diff
@@ -13,7 +13,7 @@
       display: inline-block;
       content: ' ';
       text-align: center;
-      border: 2px solid $neutral_dark;
+      border: 2px solid var(--neutral-base-black);
```
Comment (kelbyhawn) [inline apps/src/componentLibrary/checkbox/checkbox.module.scss:16]: black

File: apps/src/componentLibrary/checkbox/checkbox.module.scss [hunk 2]
```diff
@@ -28,31 +28,31 @@
 
     &:checked + i::before {
       content: "\f00c";
-      color: $neutral_white;
-      background: $light_primary_500;
-      border: 2px solid $light_primary_500;
+      color: var(--neutral-base-white);
+      background: var(--brand-teal-50);
+      border: 2px solid var(--brand-teal-50);
```
Comment (kelbyhawn) [inline apps/src/componentLibrary/checkbox/checkbox.module.scss:33]: teal

File: apps/src/componentLibrary/checkbox/checkbox.module.scss [hunk 3]
```diff
@@ -81,49 +81,49 @@
...
+      border-color: var(--neutral-gray-20);
     }
 
     input[type="checkbox"] {
       &:checked + i::before,
       &:indeterminate + i::before {
-        background: $neutral_dark20;
-        color: $neutral_white;
+        background: var(--neutral-gray-20);
+        color: var(--neutral-base-white);
```
Comment (kelbyhawn) [inline apps/src/componentLibrary/checkbox/checkbox.module.scss:113]: gray

**Review by levadadenys — APPROVED** [review]
LGTM!

**Pull request merged**